### ISSUE TITLE
Combined Yaw & Roll option for Gyro Steering Axis

### ISSUE THREAD

**Issue by Zebrazilla** [Valkirie/HandheldCompanion#97]
In perusing how the gyro works on Steam Deck I discovered that they have the option for combined yaw & roll when steering/turning. When enabled there also is the possibility to tune the contribution/multiplier applied to turning from either yaw or roll.

![image](https://user-images.githubusercontent.com/6796156/162596574-bac8b909-2331-4e4b-b1e0-e0b3df8ec1e5.png)

I would like to request it to be added to Handheld Companion so that it's possible to choose profile settings. Since I don't have a Steam Deck, nor have tried it, I have no idea if this option is any good or if setting the contribution factor of the axis is important. This does seem to me to make sense as an option though and like it has the potential to make the turning feel even better.

Maybe it also would be possible to use steam input settings to enable this for the virtual controller supplied by Handheld Companion's controller service, to see how it works and especially feels. I haven't done so yet personally as I tend to use UMC exclusively.

_Thank you for an amazing piece of software, I'm using this on Surface Go 3 paired with an Ipega PG-9083S telescopic controller and it works perfectly._

**Comment by Valkirie** [comment]
Thanks for the suggestion. I'm sure we can get this implemented. Am I reading correctly you're using HC on a Surface Go 3 ?
If yes, may I ask you to [download the attached](https://www.dropbox.com/t/HCFJ9sOvK7Lds5Hz), start it and share a screenshot ?
![image](https://user-images.githubusercontent.com/934757/162606815-dafb3163-6739-47a2-b9be-090adc815657.png)

**Comment by CasperH2O** [comment]
This is indeed an interesting feature. I saw it come by some time ago in a gyro/motion discussion. 

I wonder about it's purpose, it feels like its a form of tilt compensation for the way the user is holding a device, if that's the case I've been contemplating an automatic way of doing that. If not, I'd need some more info on a use case. Implementation wise this doesn't seem to challenging.

**Comment by Zebrazilla** [comment]
> Thanks for the suggestion. I'm sure we can get this implemented. Am I reading correctly you're using HC on a Surface Go 3 ? If yes, may I ask you to [download the attached](https://www.dropbox.com/t/HCFJ9sOvK7Lds5Hz), start it and share a screenshot ?

@Valkirie Yes that's correct, I'm using HC on a Surface Go 3 😁 I have never before used gyro in this sense, not counting messing around with a wiimote on PC many years ago, so I'm not sure how it compares to anything. Going by feel though it does feel perfect on the Surface. I'm very glad this software exists and that it makes use of the Windows API motion data!

I took a screenshot using the utility you shared, unfortunately it doesn't seem to reveal the specific sensor in use.
![image](https://user-images.githubusercontent.com/6796156/162617816-410bad0c-4c64-49e9-abc5-1392f9b36463.png)
Could not see anything that would pinpoint the name of the sensor in the device manager. Seems like M$ has obscured this or didn't think it important to divulge. I couldn't find anything online either that would shed light on the sensor used, but maybe there is something out there that I missed.

> This is indeed an interesting feature. I saw it come by some time ago in a gyro/motion discussion.
> 
> I wonder about it's purpose, it feels like its a form of tilt compensation for the way the user is holding a device, if that's the case I've been contemplating an automatic way of doing that. If not, I'd need some more info on a use case. Implementation wise this doesn't seem to challenging.
> 
> @Zebrazilla thanks for sharing the device you are using, had no idea people were using this outside the Ayaneo group. If you could provide some of the info @Valkirie asked we'd appreciate it.

@CasperH2O I saw the combined yaw & roll option in a video where someone was setting up gyro on a Steam Deck, where this option seemed to be default in the template used. I wondered about it's purpose as well and my thinking was in line with yours, I think. I'm also inclined to think that it would help aim in handheld mode as there seems to be a natural tendency to both make use of yaw and roll in the same motion in a handheld, but I really am not sure. Looking around there is very sparse information about it and it seems to be a newer addition to Steam input system, which leads me to think they included it specifically for the handheld nature of a Steam Deck.

It might be a good idea if someone knows someone with a Steam Deck to ask how it feels in comparison to just plain yaw or roll, I unfortunately don't know anyone to ask. I'm thinking it maybe is possible to try how it feels if forcing regular Steam into Steam Deck mode, as seems possible, but I haven't had a chance to try yet.
